### PR TITLE
Add test cases to verify hy-AM (Armenian-Armenia) 

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [ ] At least one test case is included for this feature or bug fix.
* [x] `ReleaseNotes` has added or is not needed.
* [x] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [x] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
* Added test cases to verify hy-AM (Armenian-Armeni)
* timezone: Asia/Yerevan (https://www.zeitverschiebung.net/en/timezone/asia--yerevan)
  When I checked the `Asia/Yerevan` timezone, it said, there's **no DaylightSaving Time**. but the ilib data returns the `+04/05` value. I'm not sure if it's correct ilib data or not.
* Added test case feature: LocaleInfo,Script, String(plural) LocaleMatch, DateFmt, DurFmt, DateRngFmt, List, Unit, firstDayofWeek, clock, calendar,Month/Weekday Translation, NumFmt,

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

File: js/test/root/testlocalematch.js [hunk 1]
```diff
@@ -1106,6 +1106,39 @@ module.exports.testlocalematch = {
...
+        test.ok(typeof(lm) !== "undefined");
+        var locale = lm.getLikelyLocale();
+        test.ok(typeof(locale) !== "undefined");
+        test.equal(locale.getSpec(), "hy-Armn-AM");
+        test.done();
+    },
+    testLocaleMatcherGetLikelyLocaleByLocaleCodehy3: function(test) {
+        test.expect(3);
+        var lm = new LocaleMatcher({
+            locale: "Aram-AM"
```
Comment (ehoogerbeets) [inline js/test/root/testlocalematch.js:1134]: Armn? Aram is not a valid script code...

File: js/test/root/testlocalematch.js [hunk 2]
```diff
@@ -1835,5 +1868,28 @@ module.exports.testlocalematch = {
...
+        test.ok(typeof(lm) !== "undefined");
+        var locale = lm.getLikelyLocaleMinimal();
+        test.ok(typeof(locale) !== "undefined");
+        test.equal(locale.getSpec(), "hy-AM");
+        test.done();
+    },
+    testLocaleMatcherGetLikelyLocaleMinimalNonDefaultLocalehyAM2: function(test) {
+        test.expect(3);
+        var lm = new LocaleMatcher({
+            locale: "hy-Aram"
```
Comment (ehoogerbeets) [inline js/test/root/testlocalematch.js:1886]: Same here

File: js/test/units/testunitfmt_hy_AM.js [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+            amount: 2000
+        });
+
+        var uf = new UnitFmt({
+            locale: "hy-AM",
+            autoConvert: true,
+            length: "long"
+        });
+        var str = uf.format(m1);
+        test.equal(str, "3,21868 megameters");
```
Comment (ehoogerbeets) [inline js/test/units/testunitfmt_hy_AM.js:220]: "Megameters" in English?

File: js/test/daterange/testSuite.html [hunk 4]
```diff
@@ -64,6 +64,7 @@
     <script src="testdatefmtrange_hi_IN.js"></script>
     <script src="testdatefmtrange_hr_HR.js"></script>
     <script src="testdatefmtrange_hu_HU.js"></script>
+    <script src="testdatefmtrange_hy_AM.js"></script>
```
Comment (ehoogerbeets) [inline js/test/daterange/testSuite.html:67]: Update file copyright year in the header

File: js/test/daterange/testSuiteFiles.js [hunk 5]
```diff
@@ -52,6 +52,7 @@ module.exports.files = [
     "testdatefmtrange_hi_IN.js",
     "testdatefmtrange_hr_HR.js",
     "testdatefmtrange_hu_HU.js",
+    "testdatefmtrange_hy_AM.js",
```
Comment (ehoogerbeets) [inline js/test/daterange/testSuiteFiles.js:55]: Update copyright year

File: js/test/date/testdatetimefmt.js [hunk 6]
```diff
@@ -5128,6 +5128,32 @@ module.exports.testdatetimeformat = {
         result1 = new DateFmt({locale:"ka-GE", type:"time", time:"ahmsz", length:"full", useNative:false, timezone:"local"}).template;
         test.equal(result1,  "HH:mm:ss z");
 
+        test.done();
```
Comment (ehoogerbeets) [inline js/test/date/testdatetimefmt.js:5131]: Update copyright year in the file header

File: js/test/date/testWeekdayTranslation.js [hunk 7]
```diff
@@ -11222,6 +11222,77 @@ module.exports.testWeekdayTranslation = {
         test.equal(value[5], "პ");
         test.equal(value[6], "შ");
 
+        test.done();
```
Comment (ehoogerbeets) [inline js/test/date/testWeekdayTranslation.js:11225]: Update copyright year in file header

File: js/test/date/testSuiteFiles.js [hunk 8]
```diff
@@ -62,6 +62,7 @@ module.exports.files = [
     "testdatefmt_he_IL.js",
     "testdatefmt_hi_IN.js",
     "testdatefmt_hr_HR.js",
+    "testdatefmt_hy_AM.js",
```
Comment (ehoogerbeets) [inline js/test/date/testSuiteFiles.js:65]: Update copyright year in file header

File: js/test/date/testMonthTranslation.js [hunk 9]
```diff
@@ -7315,6 +7315,51 @@ module.exports.testmonthtranslation = {
         test.equal(value[10], "ნოე");
         test.equal(value[11], "დეკ");
 
+        test.done();
```
Comment (ehoogerbeets) [inline js/test/date/testMonthTranslation.js:7318]: Update copyright year in file header for all modified files.

File: js/test/root/testlocalematch.js (hunk 1 above)
Comment (gouniLee) [reply js/test/root/testlocalematch.js]: I fixed it

File: js/test/units/testunitfmt_hy_AM.js (hunk 3 above)
Comment (gouniLee) [reply js/test/units/testunitfmt_hy_AM.js]: There was no translation data in hy/sysres.json. that's why it returns english. so I added it by using google translate. the default system is `metric` in hy-AM. so I added an additional test case with `meter`.

File: js/test/date/testMonthTranslation.js (hunk 9 above)
Comment (gouniLee) [reply js/test/date/testMonthTranslation.js:7318]: All modified files' copyright year has updated

File: js/test/units/testunitfmt_hy_AM.js (hunk 3 above)
Comment (ehoogerbeets) [reply js/test/units/testunitfmt_hy_AM.js]: Maybe just not test that particular case if it is not in CLDR. Only test the meters, or other case which is in CLDR.

Comment (gouniLee) [reply js/test/units/testunitfmt_hy_AM.js]: I updated it

**Comment by ehoogerbeets** [comment]
Also, please update the year in the copyright notice for each file you changed to include 2020.

**Comment by gouniLee** [comment]
> Also, please update the year in the copyright notice for each file you changed to include 2020.

When reviewing, I hope you ignore the copyright year regarding modified files.  If there is anything missing, I will check again after all PRs are merged successfully by fixing all conflicts

**Pull request merged**